src/telemetry/metrics/transport.rs

```diff
@@ -27,32 +27,28 @@ metrics! {
 /// Implements `fmt::Display` to render prometheus-formatted metrics for all transports.
 #[derive(Debug, Default)]
 pub struct Transports {
-    opens: OpenScopes,
-    closes: CloseScopes,
+    metrics: IndexMap<Key, Metrics>,
 }
 
-type OpenScopes = Scopes<TransportLabels, OpenMetrics>;
+/// Describes the dimensions across which transport metrics are aggregated.
+///
+/// Implements `fmt::Display` to render a comma-separated list of key-value pairs.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+struct Key {
+    direction: Direction,
+    peer: Peer,
+    tls_status: TlsStatus,
+}
 
+/// Holds all of the metrics for a class of transport.
 #[derive(Debug, Default)]
-struct OpenMetrics {
+struct Metrics {
     open_total: Counter,
     open_connections: Gauge,
     write_bytes_total: Counter,
     read_bytes_total: Counter,
```

Comment:
I believe the read/written bytes stats used to be labeled with the class of EOS as well? Not sure if this matters, but it is technically a change in behaviour that _could_ be breaking.

Reply:
they were not.